packages/server/src/socket/manager.ts

```diff
@@ -7,17 +7,38 @@ import { SocketService } from './service'
 
 export class SocketManager {
   private logger = new Logger('Socket')
+  private ws: Socket.Server | undefined
   private handlers: { [type: string]: SocketHandler } = {}
 
   constructor(private sockets: SocketService) {}
 
   async setup(server: Server) {
     if (yn(process.env.ENABLE_EXPERIMENTAL_SOCKETS)) {
-      const ws = new Socket.Server(server, { cors: { origin: '*' } })
-      ws.on('connection', this.handleSocketConnection.bind(this))
+      this.ws = new Socket.Server(server, { cors: { origin: '*' } })
+      this.ws.on('connection', this.handleSocketConnection.bind(this))
     }
   }
 
+  async destroy() {
+    if (this.ws !== undefined) {
+      await new Promise((resolve, reject) => {
+        // This is kind of hack to make sure that socket.io does not
+        // try to close the HTTP server before http-terminator
+        this.ws!['httpServer'] = undefined
```

Comment:
Did this happen while testing?

Reply:
Yes! Calling close on the server when it is stopped throws an unexpected exception. Since socket.io calls close on the server if it is defined (https://github.com/socketio/socket.io/blob/master/lib/index.ts#L652) and then http-terminator does the same, an exception is thrown.

Reply:
Ok cool looking at the code this should work perfeclty